metadata-editor/app/lib/MetadataMessageConsumer.scala

```diff
@@ -0,0 +1,34 @@
+package lib
+
+import play.api.libs.json._
+import com.gu.mediaservice.lib.aws.MessageConsumer
+import com.gu.mediaservice.model.Image
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.Future
+
+object MetadataMessageConsumer extends MessageConsumer(
+  Config.queueUrl, Config.awsEndpoint, Config.awsCredentials, MetadataEditorMetrics.processingLatency) {
+
+  override def chooseProcessor(subject: String): Option[JsValue => Future[Any]] =
+    PartialFunction.condOpt(subject) {
+      case "image-indexed" => processIndexedImage
+    }
+
+  def isArchivingCategory(category: String) = {
+    val archiveCategory = List("staff-photographer", "contract-photographer", "commissioned-photographer")
```

Comment:
Much nicer!

The `case _:T1 | _:T2 => foo()` syntax looks to be pretty common and readable too.